diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,183 @@
+        """
+        Load studio view in workbench
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/studio_{}.xml".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        return self.go_to_view("studio_view")
+
+
+class StudentBaseTest(DiagnosticFeedbackBaseTest):
+
+    def _get_startover_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[3]
+
+    def _get_startover_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#cancel']")
+
+    def _is_any_choice_selected(self):
+        selected = False
+        choices = [c for c in self.browser.find_elements_by_xpath("//input[@type='radio']") if c.is_displayed()]
+
+        for choice in choices:
+            if choice.is_selected():
+                selected = True
+                break
+
+        return selected
+
+    def _get_choice(self, idx):
+        return [c for c in self.browser.find_elements_by_xpath("//input[@type='radio']") if c.is_displayed()][idx]
+
+    def _verify_question(self, question_order, next_btn, back_btn, choice_idx):
+
+        self.assertEqual(self._is_any_choice_selected(), False)
+
+        if question_order > 0:
+            self.assertEqual(back_btn.get_attribute('class'), '')
+        else:
+            self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+
+        self.assertEqual(next_btn.get_attribute('class'), 'disabled')
+
+        choice = self._get_choice(choice_idx)
+        choice.click()
+        self.assertEqual(self._is_any_choice_selected(), True)
+
+        if question_order == 0:
+            self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+        else:
+            self.assertEqual(back_btn.get_attribute('class'), '')
+
+        self.assertEqual(next_btn.get_attribute('class'), '')
+
+    def _verify_startover(self, next_btn, back_btn, next_btn_link, back_btn_link, startover_link):
+        startover_link.click()
+
+        time.sleep(50)
+        self.wait_until_visible(next_btn)
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self.assertEqual(startover_link.is_displayed(), False)
+        self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+        self.assertEqual(next_btn.get_attribute('class'), 'disabled')
+
+        self._verify_question(0, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._get_previous_button_link().click()
+
+        self.assertEqual(back_btn.get_attribute('class'), '')
+        self.assertEqual(next_btn.get_attribute('class'), '')
+
+        self._get_previous_button_link().click()
+
+        self.assertEqual(back_btn.get_attribute('class'), 'disabled')
+        self.assertEqual(next_btn.get_attribute('class'), '')
+
+        self._submit_question(is_last=True)
```

Comment:
@jareerahsan I wouldn't necessarily say that `is_last` was a poor choice for the name of the first parameter of `_submit_question`. My question was more about whether you should be passing `True` or `False` here (sorry if that wasn't clear). A potential disadvantage of using `wait_until_next_btn_disabled` is that even though it does tell the reader *what* `_submit_question` is going to do, it doesn't really specify *why* it should do this. `_verify_startover` contains quite a long sequence of steps; it might be worth adding some comments explaining the sequence of actions.